shared/serial_port_controller.py

```diff
@@ -10,47 +10,29 @@
 '''
 
 import serial.tools.list_ports
-import serial
 import os
+import serial
 
 class SerialPortController():
     '''Serial Port control functions.'''
-    def __init__(self, setting_file = None):
+    def __init__(self, setting_file=None):
         self.ports_in_used = []
-
         self.baud_rate = 9600
         self.byte_size = None
         self.parity = None
         self.stop_bits = serial.STOPBITS_ONE
         self.flow_control = None
-
         self.writer_port = None
         self.reader_port = None
 
         if setting_file:
-            if os.name == 'posix':
-                file_path = os.getcwd() + "/settings/serial ports/preference/" + setting_file
-            else:
-                file_path = os.getcwd() + "\\settings\\serial ports\\preference\\" + setting_file
-            file = open(file_path, "r")
-            file_names = []
-            for line in file:
-                file_names.append(line)
-            if os.name == 'posix':
-                setting_file_path = os.getcwd() + "/settings/serial ports/" + file_names[0]
-            else:
-                setting_file_path = os.getcwd() + "\\settings\\serial ports\\" + file_names[0]
-            with open(setting_file_path, "r") as file:
-                for line in file:
-                    self.retrieve_setting([line[0], line[3], line[1], line[4], line[2]])
-                    # Whenever automation is implemented, add the serial port here
-                    # self.set_reader_port(line[6])
-
-            # settings = [baud rate, data bits/byte size, parity, stop bits, flow control option]
-            # file format: baud rate, parity, flow control option, data bits, stop bits, input method (not needed), port
-
-
-
+            # Dynamically construct the file path
```

Comment:
This is a much more elegant solution to finding the absolute paths for the local files. Looks good!